src/native/http_request_response.c

```diff
@@ -110,23 +107,12 @@ static struct http_stream_callback_data *http_stream_callback_alloc(JNIEnv *env,
     (void)jvmresult;
     AWS_FATAL_ASSERT(jvmresult == 0);
 
-    aws_mutex_init(&callback->setup_lock);
-
     callback->java_http_response_stream_handler = (*env)->NewGlobalRef(env, java_callback_handler);
```

Comment:
every CRT has some mechanism that keeps the stream object alive from `creation -> stream_complete`
It needs to change to keep the stream alive from `activate -> stream_complete` instead

I'm not sure exactly how it works in java. you'll need to figure it out